src/components/map/wr-map-legend.tsx

```diff
@@ -1,28 +1,27 @@
+import { MapPin } from "lucide-react";
+
+import type { WrLegendItem } from "@/lib/types";
 import { cn } from "@/lib/utils";
 
 interface LegendProps {
-  items: {
-    text: string;
-    color: string;
-  }[];
+  items: WrLegendItem[];
   className?: string;
 }
 
-export function Legend({ items, className }: LegendProps) {
+export function WrMapLegend({ items, className }: LegendProps) {
   return (
     <div
       className={cn(
-        "mx-auto grid w-[90%] grid-cols-2 gap-x-28 gap-y-2 sm:w-fit sm:gap-x-4",
+        "mx-auto grid w-[90%] grid-cols-2 gap-x-5 gap-y-2 sm:grid-cols-[repeat(3,minmax(0,20vw))] md:w-max",
         className,
       )}
     >
       {items.map((item) => (
-        <div key={item.text} className="flex items-center">
-          <div
-            className="mr-2 h-4 w-4 rounded-full"
-            style={{ backgroundColor: item.color }}
-          />
-          <span>{item.text}</span>
+        <div key={item.name} className="flex items-center">
+          <MapPin className="mr-2 h-8 w-8" color={item.color} />
+          <p className="w-[75%] text-wrap text-[3vw] sm:text-base">
```

Comment:
moim zdaniem dodajmy zabezpieczenie jednak przed za duza liczba linijek
```suggestion
          <p className="w-[75%] text-wrap text-[3vw] sm:text-base line-clamp-2 overflow-hidden text-ellipsis">
```